source/MaterialXGenGlsl/Nodes/TransformNormalNodeGlsl.cpp

```diff
@@ -12,6 +12,20 @@ ShaderNodeImplPtr TransformNormalNodeGlsl::create()
     return std::make_shared<TransformNormalNodeGlsl>();
 }
 
+void TransformNormalNodeGlsl::emitFunctionCall(const ShaderNode& node, GenContext& context, ShaderStage& stage) const
```

Comment:
This appears to be an incomplete implementation. This derives from a general transform from space A to space B which does not appear to be handled anywhere here.

I'd suggest injecting normalization code after the regular vector transform code.

Reply:
The first line of this function calls the base implementation of `TransformVectorNodeGlsl`, transforming the vector. It is then normalized, like you are suggesting.